lib/dialogs.dart

```diff
@@ -144,3 +146,59 @@ class SharingDialog extends DDialog {
     });
   }
 }
+
+class SettingsDialog extends DDialog {
+
+  Map keyMap;
+
+  Map get optionMap => options.values;
+
+  SettingsDialog(this.keyMap) {
+    element.classes.toggle('settings-dialog', true);
+    var text = new ParagraphElement();
+
+    text.text = 'Sharing this pad will create a permanent, publicly visible '
+    'copy on gist.github.com.';
+
+  }
+
+  DListElement get keyMapToHtml {
+    DListElement dl = new DListElement();
+    keyMap.forEach((action, keys) {
+      String string = "";
+      keys.forEach((key){
+        if (makeKeyPresentable(key) != null) {
+        string += "<span>${makeKeyPresentable(key)}</span>";
+      }
+    });
+      dl.innerHtml += "<dt>$action</dt><dd>${string}</dd>";
+    });
+    return dl;
+  }
+
+  DListElement get optionMapToHtml {
+    DListElement dl = new DListElement();
+    optionMap.forEach((key, value) {
+      dl.innerHtml += "<dt>${capitalize(key.replaceAll("_"," "))}</dt>"
+      '<dd><input type="checkbox" id="$key" ${options.getValueBool(key) ? "checked" : ""}></dd>';
+    });
+    return dl;
+  }
+
+  String capitalize(String s) => '${s[0].toUpperCase()}${s.substring(1)}';
+
+
+  void show() {
+    super.show();
+    element.children = [];
```

Comment:
We should instead add the content to `content` to get a similar look to the other dialogs.